### PR TITLE
add menu improvements

### PR DESCRIPTION
**Pull request by binary1230**
- add __getitem__() to MenuItem for ease of use with plugins
- simlpify and rename access filtering code (thanks Vicki)
- remove extra space in base.html

addressing a couple minor things from https://github.com/magfest/ubersystem/pull/1857

### CODE REVIEW

**Comment by binary1230** [comment]
also tested this out and access filtering works great, so, happy with these (and previous) changes

File: uber/menu.py [hunk 1]
```diff
@@ -51,27 +51,26 @@ def render_items_filtered_by_current_access(self):
...
-        allowed = True
-        if self.access:
-            allowed = False
-            for access_level in listify(self.access):
-                if access_level in sa.AdminAccount.access_set():
-                    allowed = True
-                    break
-
-        if not allowed:
+        if self.access and set(listify(self.access)).isdisjoint(sa.AdminAccount.access_set()):
```
Comment (binary1230) [inline uber/menu.py]: @kitsuta with the excellent one-liner.  when writing that other code I was thinking 'I bet there's some python trickery to do this in one line'

File: uber/menu.py [hunk 2]
```diff
@@ -51,27 +51,26 @@ def render_items_filtered_by_current_access(self):
...
 
         out['name'] = self.name
         if self.submenu:
             out['submenu'] = []
-            for submenu in self.submenu:
-                out['submenu'].append(submenu.render_items_filtered_by_current_access())
+            for menu_item in self.submenu:
+                filtered_menu_items = menu_item.render_items_filtered_by_current_access()
+                if filtered_menu_items:
+                    out['submenu'].append(filtered_menu_items)
```
Comment (binary1230) [inline uber/menu.py]: renamed some stuff here and exclude None items, which makes the HTML rendering a little nicer.

File: uber/menu.py [hunk 3]
```diff
@@ -51,27 +51,26 @@ def render_items_filtered_by_current_access(self):
...
         else:
             out['href'] = self.href
 
         return out
 
+    def __getitem__(self, key):
+        for sm in self.submenu:
+            if sm.name == key:
+                return sm
+
```
Comment (binary1230) [inline uber/menu.py]: this ```__getitem__``` is optional but useful for plugins to be able to say stuff like this:
```
c.MENU['People'].append_menu_item(
    MenuItem(access=c.BANDS, name='Bands', href='../band_admin/')
)
```

some open PRs are in the plugins here:
https://github.com/magfest/bands/pull/45

File: uber/templates/base.html [hunk 4]
```diff
@@ -169,7 +169,7 @@
 
                                 <li class="dropdown">
                                     <a href="#" class="dropdown-toggle" data-toggle="dropdown">
-                                        {{  menu_item.name }}
```
Comment (binary1230) [inline uber/templates/base.html]: remove this space

File: uber/menu.py (hunk 1 above)
Comment (kitsuta) [inline uber/menu.py]: Oh jeez, I hope you tested this. :P

**Comment by binary1230** [comment]
I tested it :)

**Pull request merged**